### PR TITLE
Move `CloudList.setOwner` call from `Jenkins.loadTasks` to `load`

### PR DESCRIPTION
**Pull request by jglick**
In CloudBees CI we noticed an issue whereby after reloading `$JENKINS_HOME/config.xml` via `Jenkins.load()` (exposed as `Loadable` in #8544), `Jenkins.clouds` had a blank (`NOOP`) `owner`. This meant that subsequent changes to the cloud list were not saved. `loadTasks` (which calls `load` in its `loadJenkins` task) fixed up the owner but just `load` did not.

### Testing done

Covered by a regression test in CloudBees CI. I could try to construct something analogous here though I am not sure how compelling it would be out of context.

### Proposed changelog entries

- Setting a proper owner for `Jenkins.clouds` after `Jenkins.load()`.

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by julieheard — APPROVED** [review]

**Review by Vlatombe — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
